File: java/Samples/BiometricStandards/CS/ANTemplateSampleCS/RecordCreateForms/ANType1RecordCreateForm.cs
using System;
using System.ComponentModel;
using System.Windows.Forms;
using Neurotec.Biometrics.Standards;

namespace Neurotec.Samples.RecordCreateForms
{
	public partial class ANType1RecordCreateForm : ANRecordCreateForm
	{
		#region Public constructor

		public ANType1RecordCreateForm()
		{
			InitializeComponent();

			nudIdc.Enabled = false;
			nudIdc.Minimum = -1;
			nudIdc.Value = -1;

			Array values = Enum.GetValues(typeof(ANValidationLevel));
			foreach (object item in values)
			{
				cbValidationLevel.Items.Add(item);
			}
			cbValidationLevel.SelectedIndex = 0;
		}

		#endregion

		#region Public properties

		public string TransactionType
		{
			get { return tbTransactionType.Text; }
			set { tbTransactionType.Text = value; }
		}

		public string DestinationAgency
		{
			get { return tbDestinationAgency.Text; }
			set { tbDestinationAgency.Text = value; }
		}

		public string OriginatingAgency
		{
			get { return tbOriginatingAgency.Text; }
			set { tbOriginatingAgency.Text = value; }
		}

		public string TransactionControl
		{
			get { return tbTransactionControlId.Text; }
			set { tbTransactionControlId.Text = value; }
		}

		public ANValidationLevel ValidationLevel
		{
			get { return (ANValidationLevel)cbValidationLevel.SelectedItem; }
			set { cbValidationLevel.SelectedItem = value; }
		}

		public bool UseNistMinutiaNeighbors
		{
			get { return chbUseNistMinutiaNeighbors.Checked; }
			set { chbUseNistMinutiaNeighbors.Checked = value; }
		}

		public bool UseTwoDigitIdc
		{
			get { return chbUseTwoDigitIdc.Checked; }
			set { chbUseTwoDigitIdc.Checked = value; }
		}

		public bool UseTwoDigitFieldNumber
		{
			get { return chbUseTwoDigitFieldNumber.Checked; }
			set { chbUseTwoDigitFieldNumber.Checked = value; }
		}

		public bool UseTwoDigitFieldNumberType1
		{
			get { return chbUseTwoDigitFieldNumberType1.Checked; }
			set { chbUseTwoDigitFieldNumberType1.Checked = value; }
		}

		#endregion

		#region Private form events

		private void TbTransactionTypeValidating(object sender, CancelEventArgs e)
		{
			string value = ((TextBox)sender).Text;

			if (value.Length < ANType1Record.MinTransactionTypeLengthV4 || value.Length > ANType1Record.MaxTransactionTypeLengthV4)
			{
				errorProvider.SetError((TextBox)sender,
					string.Format("Transaction type value must be {0} to {1} characters long",
					ANType1Record.MinTransactionTypeLengthV4, ANType1Record.MaxTransactionTypeLengthV4));
				e.Cancel = true;
			}
			else
			{
				errorProvider.SetError((TextBox)sender, null);
			}
		}

		private void CbValidationLevelSelectedIndexChanged(object sender, EventArgs e)
		{
			bool isStandard = ValidationLevel == ANValidationLevel.Standard;
			tbDestinationAgency.Enabled = tbOriginatingAgency.Enabled = tbTransactionControlId.Enabled = tbTransactionType.Enabled = isStandard;
		}

		#endregion
	}
}

File: java/Samples/BiometricStandards/VB/ANTemplateSampleVB/RecordCreateForms/ANType1RecordCreateForm.vb
Imports System
Imports System.ComponentModel
Imports System.Windows.Forms
Imports Neurotec.Biometrics.Standards

Namespace RecordCreateForms

	Partial Public Class ANType1RecordCreateForm
		Inherits ANRecordCreateForm
#Region "Public constructor"

		Public Sub New()
			InitializeComponent()

			nudIdc.Enabled = False
			nudIdc.Minimum = -1
			nudIdc.Value = -1

			Dim values As Array = System.Enum.GetValues(GetType(ANValidationLevel))
			For Each item As Object In values
				cbValidationLevel.Items.Add(item)
			Next item
			cbValidationLevel.SelectedIndex = 0
		End Sub

#End Region

#Region "Public properties"

		Public Property TransactionType() As String
			Get
				Return tbTransactionType.Text
			End Get
			Set(ByVal value As String)
				tbTransactionType.Text = value
			End Set
		End Property

		Public Property DestinationAgency() As String
			Get
				Return tbDestinationAgency.Text
			End Get
			Set(ByVal value As String)
				tbDestinationAgency.Text = value
			End Set
		End Property

		Public Property OriginatingAgency() As String
			Get
				Return tbOriginatingAgency.Text
			End Get
			Set(ByVal value As String)
				tbOriginatingAgency.Text = value
			End Set
		End Property

		Public Property TransactionControl() As String
			Get
				Return tbTransactionControlId.Text
			End Get
			Set(ByVal value As String)
				tbTransactionControlId.Text = value
			End Set
		End Property

		Public Property ValidationLevel() As ANValidationLevel
			Get
				Return CType(cbValidationLevel.SelectedItem, ANValidationLevel)
			End Get
			Set(ByVal value As ANValidationLevel)
				cbValidationLevel.SelectedItem = value
			End Set
		End Property

		Public Property UseNISTMinutiaNeighboars() As Boolean
			Get
				Return chbUseNistMinutiaNeighbors.Checked
			End Get
			Set(ByVal value As Boolean)
				chbUseNistMinutiaNeighbors.Checked = value
			End Set
		End Property

#End Region

#Region "Private form events"

		Private Sub TbTransactionTypeValidating(ByVal sender As Object, ByVal e As CancelEventArgs) Handles tbTransactionType.Validating
			Dim value As String = (CType(sender, TextBox)).Text

			If value.Length < ANType1Record.MinTransactionTypeLengthV4 OrElse value.Length > ANType1Record.MaxTransactionTypeLengthV4 Then
				errorProvider.SetError(CType(sender, TextBox), String.Format("Transaction type value must be {0} to {1} characters long", ANType1Record.MinTransactionTypeLengthV4, ANType1Record.MaxTransactionTypeLengthV4))
				e.Cancel = True
			Else
				errorProvider.SetError(CType(sender, TextBox), Nothing)
			End If
		End Sub

		Private Sub CbValidationLevelSelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles cbValidationLevel.SelectedIndexChanged
			Dim isStandard As Boolean = ValidationLevel = ANValidationLevel.Standard
			tbTransactionType.Enabled = isStandard
			tbTransactionControlId.Enabled = tbTransactionType.Enabled
			tbOriginatingAgency.Enabled = tbTransactionControlId.Enabled
			tbDestinationAgency.Enabled = tbOriginatingAgency.Enabled
		End Sub

#End Region
	End Class
End Namespace
